docs/DeveloperGuide.md

```diff
@@ -1,38 +1,324 @@
+Manage core clinic related data faster than using mouse or GUI.
+Ensure each data type conforms to certain standards with in-built validations.
 
-{Describe the value proposition: what problem does it solve?}
+### User stories
 
-## User Stories
+Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unlikely to have) - `*`
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | Priority | As a ... | I want to ... | So that I can ... |
+|---------| -------- | ------------------------------------------ | ------------------------------ | ---------------------------------------------------------------------- |
+|         | `* * *`  | new user                                   | see usage instructions         | refer to instructions when I forget how to use the App                 |
+|         | `* * *`  | new user                                   | import or export data          | share with others when needed and make backup copies                                                                       |
+|         | `* *`    | user                                       | find appointments based on selected criteria | filter out the appointments that I want to know about                                   |
+|         | `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
+|         | `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
+|         | `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
 
-## Non-Functional Requirements
+### Use cases
 
-{Give non-functional requirements}
 
+### Non-Functional Requirements
+Device Environment:
+* Must have Java 11 or higher installed in OS
+* 32-bit or 64-bit environment
+* Command Line Interface supported
+
+  Performance of app:
+* Function offline, without the need for internet access
+* Quick to launch and use
+* No noticeable lag or delay in performance when running
+* Intuitive and seamless for new users.
+* Ability to export the data into a txt file to load on another OS
+
+  Reliability of app:
+* Data files should be updated constantly and accurately, with no data loss
+* Data records should be retrievable and readable
+* Text inputs should produce similar results if utilised multiple times.
+* Program should run without any forced-close error due to bugs
+--------------------------------------------------------------------------------------------------------------------
```

Comment:
Are the indentations here intentional? Looks inconsistent with the Device Environment portion